### PR TITLE
Fixed PCA docs and caching

### PR DESCRIPTION
**Pull request by jbloom22**
Still not the ideal (M as transpose is confusing and we should probably automatically mean-center), but at least the docs are now correct.

### CODE REVIEW

**Comment by jbloom22** [comment]
This partially addresses #2734 (still need to resolve whether to automate mean-centering)

File: python/hail/methods/statgen.py [hunk 1]
```diff
@@ -1298,8 +1299,36 @@ def hwe_normalized_pca(dataset, k=10, compute_loadings=False, as_array=False):
...
+    or all homozygous alternate are unnormalizable and removed before
+    evaluation.
+
+    Users of PLINK/GCTA should be aware that Hail computes the GRM slightly
+    differently with regard to missing data. In Hail, the
+    :math:`ij` entry of the GRM :math:`MM^T` is simply the dot product of rows
+    :math:`i` and :math:`j` of :math:`M`; in terms of :math:`C` it is
+
+    .. math:
+      \\frac{1}{m}\sum_{l\in\mathcal{C}_i\cap\mathcal{C}_j}\\frac{(C_{il}-2p_l)(C_{jl} - 2p_l)}{2p_l(1-p_l)
```
Comment (patrick-schultz) [inline python/hail/methods/statgen.py]: Any reason you're not using a raw string?

**Comment by jbloom22** [comment]
I've updated docs and code to ensure that NaN and infinite values are caught.

File: python/hail/methods/statgen.py (hunk 1 above)
Comment (jbloom22) [reply python/hail/methods/statgen.py]: going by the "if it ain't broke and there are more critical needs, don't fix it" philosophy

**Pull request merged**